tests/test_analyze_docker_dockerfile.py

```diff
@@ -102,6 +105,43 @@ def testParseFromImage(self):
                                        'digest_type': 'sha256',
                                        'digest': debian_digest}])
 
+    def testFindGitInfo(self):
+        sample_line = 'ADD test_git_repo.tar /tmp'
+        comment_line = dockerfile.find_git_info(sample_line, self.pin_add)
+        expected = ('git project name: test_repo.git '
+                    'sha info: 55cbf5cbc8b201a99ac63f70f6f6f0ff1d1ae630')
+        self.assertEqual(comment_line, expected)
+
+    def testExpandAddCommand(self):
+        dfobj = dockerfile.get_dockerfile_obj(self.pin_add)
+        self.assertFalse(dfobj.is_none())
+        dockerfile.expand_add_command(dfobj)
+        expected = [{'instruction': 'FROM',
+                     'startline': 0,
+                     'endline': 0,
+                     'content': 'FROM ubuntu\n',
+                     'value': 'ubuntu'},
+                    {'instruction': 'ADD',
+                     'startline': 1,
+                     'endline': 1,
+                     'content': ('ADD plain_file /tmp # Not a folder '
+                                 'or tar file\n'),
+                     'value': 'plain_file /tmp'},
+                    {'instruction': 'ADD',
+                     'startline': 2,
+                     'endline': 2,
+                     'content': ('ADD test_git_repo /tmp # Not a git '
+                                    'repository\n'),
+                     'value': 'test_git_repo /tmp'},
+                    {'instruction': 'ADD',
+                     'startline': 3,
+                     'endline': 3,
+                     'content': ('ADD test_git_repo.tar /tmp # git project '
+                                 'name: test_repo.git sha info: 55cbf5cbc8b'
+                                 '201a99ac63f70f6f6f0ff1d1ae630\n'),
+                     'value': 'test_git_repo.tar /tmp'}]
```

Comment:
I don't think this functionality is testable, because the git sha of the tern repo will change every time a PR is merged. We'll just have to test the parts that adds the comment and the function that gets the git repository name (if those two things are in different functions).